incubator/command-config/src/main/java/io/aklivity/zilla/runtime/command/config/internal/openapi/http/proxy/OpenApiHttpProxyConfigGenerator.java

```diff
@@ -0,0 +1,505 @@
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package io.aklivity.zilla.runtime.command.config.internal.openapi.http.proxy;
+
+import static io.aklivity.zilla.runtime.binding.http.config.HttpPolicyConfig.CROSS_ORIGIN;
+import static io.aklivity.zilla.runtime.engine.config.KindConfig.CLIENT;
+import static io.aklivity.zilla.runtime.engine.config.KindConfig.SERVER;
+import static java.util.Objects.requireNonNull;
+import static org.agrona.LangUtil.rethrowUnchecked;
+
+import java.io.InputStream;
+import java.net.URI;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import jakarta.json.Json;
+import jakarta.json.JsonArrayBuilder;
+import jakarta.json.JsonObject;
+import jakarta.json.JsonPatch;
+import jakarta.json.bind.Jsonb;
+import jakarta.json.bind.JsonbBuilder;
+import jakarta.json.spi.JsonProvider;
+
+import io.aklivity.zilla.runtime.binding.http.config.HttpConditionConfig;
+import io.aklivity.zilla.runtime.binding.http.config.HttpOptionsConfig;
+import io.aklivity.zilla.runtime.binding.http.config.HttpOptionsConfigBuilder;
+import io.aklivity.zilla.runtime.binding.tcp.config.TcpConditionConfig;
+import io.aklivity.zilla.runtime.binding.tcp.config.TcpOptionsConfig;
+import io.aklivity.zilla.runtime.binding.tls.config.TlsOptionsConfig;
+import io.aklivity.zilla.runtime.command.config.internal.airline.ConfigGenerator;
+import io.aklivity.zilla.runtime.command.config.internal.openapi.model.OpenApi;
+import io.aklivity.zilla.runtime.command.config.internal.openapi.model.Server;
+import io.aklivity.zilla.runtime.command.config.internal.openapi.model2.PathItem2;
+import io.aklivity.zilla.runtime.command.config.internal.openapi.model2.Server2;
+import io.aklivity.zilla.runtime.engine.config.BindingConfigBuilder;
+import io.aklivity.zilla.runtime.engine.config.ConfigWriter;
+import io.aklivity.zilla.runtime.engine.config.NamespaceConfig;
+import io.aklivity.zilla.runtime.engine.config.NamespaceConfigBuilder;
+import io.aklivity.zilla.runtime.engine.config.RouteConfigBuilder;
+import io.aklivity.zilla.runtime.guard.jwt.config.JwtOptionsConfig;
+import io.aklivity.zilla.runtime.vault.filesystem.config.FileSystemOptionsConfig;
+
+public class OpenApiHttpProxyConfigGenerator implements ConfigGenerator
+{
+    private final InputStream inputStream;
+
+    private OpenApi openApi;
+    private int[] allPorts;
+    private int[] httpPorts;
+    private int[] httpsPorts;
+    private boolean isPlainEnabled;
+    private boolean isTlsEnabled;
+    private Map<String, String> securitySchemes;
+    private boolean isJwtEnabled;
+
+    public OpenApiHttpProxyConfigGenerator(
+        InputStream inputStream)
+    {
+        this.inputStream = inputStream;
+    }
+
+    public String generate()
+    {
+        this.openApi = parseOpenApi(inputStream);
```

Comment:
Does `openApi` need to be a field, or can it be passed to other methods resolving lookups?